src/test/java/uk/gov/hmcts/reform/slc/services/servicebus/MessageProcessorTest.java

```diff
@@ -24,24 +22,23 @@
 
     private MessageProcessor processor;
 
-    private Function<IMessage, MessageHandlingResult> action;
-
     @Mock private IMessageReceiver messageReceiver;
+    @Mock private SendLetterService sendLetterService;
     @Mock private IMessage message;
 
     @Before
     public void setUp() throws Exception {
-        this.processor = new MessageProcessor(() -> messageReceiver);
+        this.processor = new MessageProcessor(() -> messageReceiver, sendLetterService);
     }
 
     @Test
     public void should_complete_message_if_passed_function_returns_success() throws Exception {
         // given
-        action = (msg) -> SUCCESS;
+        given(sendLetterService.send(any())).willReturn(SUCCESS);
         given(messageReceiver.receive()).willReturn(message);
 
         // when
-        processor.handle(action);
+        processor.process();
 
         // then
         verify(messageReceiver, times(1)).complete(any());
```

Comment:
could you add `verifyNoMoreInteractions` for test cases

Reply:
perhaps separately, not relate to this change